src/server/trpc/router/admin/events/create.ts

```diff
@@ -0,0 +1,28 @@
+import { protectedProcedure } from '~/server/trpc/trpc'
+import { z } from 'zod'
+import { TRPCError } from '@trpc/server'
+
+export const createEvent = protectedProcedure
+  .input(
+    z.object({
+      name: z.string(),
+      date: z.date(),
+      departments: z.array(z.string()),
+      attendees: z.array(z.string()),
+    })
+  )
+  .mutation(async ({ ctx, input }) => {
+    try {
+      await ctx.prisma.event.create({
+        //create a new row in the table
+        data: {
+          name: input.name,
+          date: input.date,
+          departments: input.departments,
+          attendees: input.attendees,
+        },
+      })
+    } catch (error) {
+      console.log(error)
```

Comment:
- [ ] Use `console.error((error as Error).message)` instead of `console.log()`, 
- [ ] If you want, you can throw a TRPCError with the corresponding code, description and message. Otherwise, remove the import from line 3.